### ISSUE TITLE
InitializeInstanceFields should be called before returning from super call

### ISSUE THREAD

**Issue by gsathya** [babel/babel#6872]
> Choose one: is this a bug report or feature request?

bug

### Input Code

```js
var log = [];

class Foo extends class {} {
  x = log.push(1);

  constructor() {
    super();
    super();
  }
}

try {
	new Foo;
} catch(e) {}

console.log(log)
```

### Babel/Babylon Configuration (.babelrc, package.json, cli command)

stage3,stage2

### Expected Behavior

`[1, 1]`

### Current Behavior

`[1]`

### Your Environment

[REPL](https://babeljs.io/repl/build/master#?babili=false&browsers=&build=&builtIns=false&code_lz=G4QwTgBANg9g5hAvBA2gXQNwCgsGMogDOhEAYjDBAKYAeALlQHYAmJ-RJA3gL4SdYQINJNHgA6AA4BXQgAsAFAEYAlNgERcMRoTpgpuOjDDzlfdYMJSJVY6vMRL129kHcsbrLoCeZgJCMqAHcyCmxeXBA6XAUqUx4cTW0YKCoxWDh5dOUgA&debug=false&circleciRepo=&evaluate=true&lineWrap=false&presets=stage-2%2Cstage-3&targets=&version=7.0.0-beta.32)

**Comment by xtuc** [comment]
ah good catch!

Does it make sense to call `super` multiple times? Otherwise we should keep a state in the transformation to know if we already initialized stuff.

**Comment by gsathya** [comment]
> Does it make sense to call super multiple times?

Nope. It's an error. But it should error out *after* the field initializers are run.

**Comment by xtuc** [comment]
Ok so we should probably transform the second super call to a throw statement.

**Comment by gsathya** [comment]
Except you still want to call the super constructor.

```
var i = 0;

class X {
  constructor() { i++; } 
}

class P extends X {
  constructor() {
    super();
    super();
  }
}

try { new P; } catch(e) {}
console.log(i);
```

This should print `2`

**Comment by nicolo-ribaudo** [comment]
@xtuc It isn't always that easy:

e.g.
```js
class A {
  constructor() {
    if (Math.random() > .5) super();
    if (Math.random() > .5) super();
  }
}
```

The problem is that `_possibleConstructorReurn` throws before the initializers are executed, while the correct order should be (to my understanding)
1. Call the super constructor
2. Run properties initializers
3. If `this` was already defined, throw an error
(currently 2 and 3 are swapped)

**Comment by gsathya** [comment]
> @gsathya Should the output be 0 or 1?

`0`

**Comment by xtuc** [comment]
Ok got it. We need to figure it out at runtime.

**Comment by thiagoarrais** [comment]
This seems to be fixed in master now. Would a PR adding this test case be interesting?

**Comment by ljharb** [comment]
Yes.
